packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.tsx

```diff
@@ -138,77 +99,31 @@ function _QueryWrapper(props: QueryWrapperProps) {
     addValueToSelectedFacet,
     setRangeFacetValue,
     resetDebounceTimer,
+    currentPage,
+    goToPage,
+    pageSize,
+    setPageSize,
   } = immutableTableQueryResult
 
   const lastQueryRequest = useMemo(() => {
     return getCurrentQueryRequest()
   }, [getCurrentQueryRequest])
 
-  const queryWrapperData = useQueryWrapperData(lastQueryRequest, isInfinite)
-
-  const { data, isLoadingNewBundle, error, currentAsyncStatus } =
-    queryWrapperData
-
-  const paginationControls = useQueryWrapperPaginationControls({
-    immutableTableQueryResult,
-    queryWrapperData,
-    isInfinite,
-  })
-
-  const { data: entity } = useGetEntity<Table>(entityId, versionNumber)
-
-  // data is sometimes undefined, which useDeepCompareEffect doesn't like, so use useDeepCompareEffectNoCheck instead
-  useDeepCompareEffectNoCheck(() => {
-    if (data && onQueryResultBundleChange) {
-      onQueryResultBundleChange(JSON.stringify(data))
-    }
-  }, [data, onQueryResultBundleChange])
-
-  const isFacetsAvailable = data
-    ? isFacetAvailable(data.facets, data.selectColumns)
-    : true
-
-  const setIsLoadingNewBundle = useSetAtom(isLoadingNewBundleAtom)
-  useEffect(() => {
-    setIsLoadingNewBundle(isLoadingNewBundle)
-  }, [isLoadingNewBundle, setIsLoadingNewBundle])
-
-  const setLockedColumn = useSetAtom(lockedColumnAtom)
-  useEffect(() => {
-    setLockedColumn(lockedColumn)
-  }, [lockedColumn, setLockedColumn])
-
-  /**
-   * Remove a particular facet name (e.g. study) and all possible values based on the parameter specified in the url
-   * this is to remove the facet from the charts, search and filter.
-   * @return data: QueryResultBundle
-   */
-  const dataWithLockedColumnFacetRemoved = useMemo(() => {
-    return removeLockedColumnFromFacetData(data, lockedColumn)
-  }, [data, lockedColumn])
-
-  const setTableQueryData = useSetAtom(tableQueryDataAtom)
-  useEffect(() => {
-    setTableQueryData(dataWithLockedColumnFacetRemoved)
-  }, [dataWithLockedColumnFacetRemoved, setTableQueryData])
+  const {
+    rowDataQueryOptions,
+    rowDataInfiniteQueryOptions,
+    queryMetadataQueryOptions,
+  } = useTableQueryUseQueryOptions(lastQueryRequest, lockedColumn)
```

Comment:
For convenience get `UseQueryOptions` for the different table query data partitions